homeassistant/helpers/condition.py

```diff
@@ -461,7 +461,13 @@ def async_template(
     if isinstance(value, bool):
         return value
 
-    return str(value).lower() == "true"
+    if isinstance(value, str):
+        return value.lower() == "true"
+
+    if isinstance(value, ResultWrapper):
```

Comment:
If we don't care about the result wrapper, why not always render to string?

Reply:
Actually, we don't have to render to string,  as the wrapper would always be a set, list, tuple or dict. Hence, this whole check is useless.